src/components/Login.jsx

```diff
@@ -47,13 +47,14 @@ export const Login = () => {
       const role = responseRole.data[0].authority + responseRole.data[1].authority ;
```

Comment:
I'm adding these comments here because I can't add comments to the code above.

This object should be somewhere else. What do you think would be a good place to put it?

```
const initialState = {
    username:'',
    password:''
}
```

..........................................................................................

I believe that `credentials` and `setCredentials` would be more intuitive because that's what these inputs are used for.

```const [loginForm,setLoginForm] = useState(initialState)```

..........................................................................................

I would put:

```
const responseToken = await axios.post('http://localhost:8080/bank/api/v1/auth', {},
        {
          params: {
            username: loginForm.username,
            password: loginForm.password
          }
        }
      )
```

and:

```
const responseRole = await axios.get('http://localhost:8080/bank/api/v1/user/role',{
        headers:{Authorization:`Bearer ${token}`}
      });
```

... in your `index.js` file.

..........................................................................................

Try the _String Interpolation_ way here: `const role = responseRole.data[0].authority + responseRole.data[1].authority`

Reply:
`const initialState = {
    username:'',
    password:''
}`

Ok, let me think where I can put this object, I put it there because the initial state should be outside the component function.

Reply:
String interpolation looks weird `const role = `${responseRole.data[0].authority} ${responseRole.data[1].authority}`` . Are you sure?

Reply:
![image](https://user-images.githubusercontent.com/111784440/195091414-7c866e1a-33ea-46b2-b9b8-b28749759ba6.png)

Reply:
Why do you need `responseRole.data[0].authority`? AFAIK, `responseRole.data[1].authority` has what you need (ROLE_USER or ROLE_ADMIN).

Reply:
![image](https://user-images.githubusercontent.com/111784440/195108803-f28bd4ff-8f64-4e17-8ded-9f6385eaa794.png)

Reply:
You never know :)

Reply:
That's interesting! Well, I believe you can either keep it as it is or create a function to determine to handle roles differently. You can use something like this: `const filteredRole = randomArray.some(o => o.authority === 'ROLE_ADMIN');` `randomArray` would be the array above.